src/Plugin/ContentTypePlugin.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Http\Client\Common\Plugin;
+
+use Http\Client\Common\Plugin;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Allow to set the correct content type header on the request automatically only if it is not set .
+ *
+ * @author Karim Pinchon <karim.pinchon@gmail.com>
+ */
+final class ContentTypePlugin implements Plugin
```

Comment:
May be nice to add some options to this plugin to avoid detection on too large stream, or stream with a null size